lib/admin/WP_Auth0_Admin_Basic.php

```diff
@@ -354,19 +357,21 @@ public function basic_validation( $old_options, $input ) {
 			return $input;
 		}
 
+		$input['domain']           = sanitize_text_field( $input['domain'] );
+		$input['custom_domain']    = sanitize_text_field( $input['custom_domain'] );
 		$input['client_id']        = sanitize_text_field( $input['client_id'] );
 		$input['cache_expiration'] = absint( $input['cache_expiration'] );
 
-		$input['allow_signup'] = ( isset( $input['allow_signup'] ) ? $input['allow_signup'] : 0 );
+		$input['client_secret'] = sanitize_text_field( $input['client_secret'] );
+		if ( __( '[REDACTED]', 'wp-auth0' ) === $input['client_secret'] ) {
```

Comment:
If client secret is unchanged (set as `[REDACTED]` when saved), keep the previous one.